sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -365,8 +387,16 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
         numEventsInWaitList, phEventWaitList, Guard, &StreamToken);
     CUfunction CuFunc = hKernel->get();
 
-    Result = enqueueEventsWait(hQueue, CuStream, numEventsInWaitList,
-                               phEventWaitList);
+    if (DepEvents.size()) {
+      Result = enqueueEventsWait(CuStream, DepEvents.size(), &DepEvents[0]);
```

Comment:
`Result` is never validated and will either be ignored by an early return, or overridden.

Reply:
Good catch, change made.